README.md

```diff
@@ -1,6 +1,394 @@
 # CNS Object Broker
-## For dynmically provisioning object storage compliant with an S3 interface.
 
 ## WARNING!
+The work in this project is a proof of concept and not intended for production.
 
-The work in this project is a proof of concept and not intended for anything more than that.
+## Utilize Kubernetes Service-Catalog to dynamically provision CNS Object Storage.
+
+## Overview
+A core feature of the Kubernetes system is the ability to provision a diverse
+offering of block and file storage on demand.
+This project seeks to demonstrate
+that by using the Kubernetes-Incubator's [Service-Catalog](https://github.com/kubernetes-incubator/service-catalog),
+it is now also possible to bring dynamic provisioning to S3 object storage.
+This is accomplished with an application called a Broker that implements the [OpenServiceBroker](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md) API spec.
+The Broker handles requests to create and destroy object buckets and returns information required to consume them.
+
+The CNS Object Broker is designed
+to be used with [Gluster-Kubernetes](https://github.com/jarrpa/gluster-kubernetes).
+Our [command flow diagram](docs/diagram/control-diag.md) shows where Kubernetes, Service-Catalog,
+and the (often external) service broker interact. Service-catalog terms used below are included in the diagram.
+
+### Nomenclature
+
+Before going further, it's useful to define common terms used throughout the entire system.
+The relationships between the different systems and objects can be seen in the [control flow diagram](docs/diagram/control-diag.md).
+There are a number of naming collisions which can lead to some confusion.
+
+#### External Service Provider
+
+- *External Service Provider*
+
+For our purposes, an External Service Provider is any system that uses a Service Broker to make Services available on demand. The actual location of the External Service Provider is arbitrary.
+It can be a remote service provider like AWS S3, an on premise cluster, or colocated with the Service-Catalog.
+
+The External Service Provider should consist of two components: the service broker and the actual services being consumed by clients.  In this example, the service component is a CNS Object Storage cluster with a Swift/S3 REST API.
+
+- *Broker*
+
+The Broker binary is run inside a pod.  The broker implements a server that listens for REST requests and translates them to the underlying Broker methods.  When methods complete, they return OpenServiceBroker responses in json format.
+
+##### Broker Objects
+
```

Comment:
Do you want to define "ServiceClass"?

Reply:
Brokers do not have a type `ServiceClass`.  In the catalog, there are `Services` and `Plans`, which are mentioned under the Catalog bullet.  I can add bullets for them as well.